src/misc.cpp

```diff
@@ -299,21 +300,94 @@ std::string compiler_info() {
 
 
 /// Debug functions used mainly to collect run-time statistics
-static std::atomic<int64_t> hits[2], means[2];
+constexpr int MaxDebugSlots = 32;
 
-void dbg_hit_on(bool b) { ++hits[0]; if (b) ++hits[1]; }
-void dbg_hit_on(bool c, bool b) { if (c) dbg_hit_on(b); }
-void dbg_mean_of(int v) { ++means[0]; means[1] += v; }
+namespace {
+
+template<int _Vars>
+struct DebugInfo {
+    std::atomic<int64_t> data[_Vars] = { 0, };
```

Comment:
superfluous `,`  and also `_Vars` is a bad name imo. it's one variable not multiple and doesnt say much,  
perhaps `size_t N` might be better as I think its commonly used in template parameters

Reply:
Fixed.